### PR TITLE
CO-1187: Add weeklyMenu field to applicable dining locations

### PR DESCRIPTION
**Pull request by ianbrown9475**

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/locations/mapper/LocationMapper.groovy [hunk 1]
```diff
@@ -132,12 +133,16 @@ class LocationMapper {
...
     }
 
-    private void setLinks(ResourceObject resourceObject) {
+    private void setLinks(ResourceObject resourceObject, String locId) {
         String resource = isService(resourceObject.attributes) ? Constants.SERVICES :
                 Constants.LOCATIONS
 
         String selfUrl = "$apiEndpointUrl$resource/${resourceObject.id}"
         def links = ['self': selfUrl]
+        if(resourceObject.attributes.type == "dining" && locId != null) {
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/locations/mapper/LocationMapper.groovy]: Could you use this constant instead? https://github.com/osu-mist/locations-api/blob/76156c43e90ce81994c17af19a44d01bc1187b3f/src/main/groovy/edu/oregonstate/mist/locations/Constants.groovy#L6

File: src/main/groovy/edu/oregonstate/mist/locations/mapper/LocationMapper.groovy [hunk 2]
```diff
@@ -132,12 +133,16 @@ class LocationMapper {
...
+            String menuLink = "${weeklyMenuUrl}?loc=${locId}"
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/locations/mapper/LocationMapper.groovy]: Could you use a Uri builder to build this string instead?

**Review by jaredkosanovic — CHANGES_REQUESTED** [review]
What do you think about putting this link in the attributes section instead? We already have some links in there anyway because they're links to other services (not ones the API provided). This way, you wouldn't have to check if every resource object is a dining object. The link could be generated apart of the mapDiningLocation method (or whatever it's called). Also, the field would always be there (but sometimes be null). The client wouldn't have to check if the location is a dining location before trying to get the menu link. What are your thoughts?

**Comment by ianbrown9475** [comment]
I agree that the link should be in the attributes field. Good to know that links should generally go here if they aren't provided by the API, I'll keep that in mind.

**Pull request merged**